app/models/role.rb

```diff
@@ -141,871 +141,768 @@ def self.permissions_for_access(access)
     permissions_with_descriptions.select { |_k, attrs| attrs[:access].include?(access) }.keys
   end
 
+  def self.permissions_by_group
+    {}.tap do |perms|
+      permissions_with_descriptions.each do |key, role|
+        perms[role[:category]] ||= {}
+        perms[role[:category]][role[:sub_category]] ||= {}
+        perms[role[:category]][role[:sub_category]][key] = role
+      end
+    end
+  end
+
   def enabled_permissions
     self.class.permissions_with_descriptions.select { |k, _| send(k) }
   end
 
+  # Pick a background color that is unique to the name, but not terribly vibrant
+  def bg_color
```

Comment:
I think these are too dark, I'm getting values where the color text contrast is definitely not accessible. I wonder about why not using the `many-colors` mixin? Since that seems to work pretty well elsewhere like for data sources. They look a bit lighter.

Reply:
Switched to many-colors mixin, thanks!